app/charts/area/areas-state.tsx

```diff
@@ -184,7 +184,7 @@ const useAreasState = (
     );
   }, [getXAsString, getY, scalesData]);
 
-  const normalize = fields.segment?.calculation === "percent";
+  const normalize = interactiveFiltersConfig?.calculation.type === "percent";
```

Comment:
Why is it in the interactiveFilters config ? It seems we could have a stacked percent bar chart without anything interactive 🤔  ?

Reply:
Honestly I wasn't 100% sure where exactly to put this (that's why initially it was inside the segment field). I thought it would be simpler to just have it inside the `interactiveFiltersConfig`, because there we do not need to do the sorts of checks "if segment exists" and then "if calculation property exists in segment" (and now we still can publish a 100%-based chart without actually enabling interactive filter [when `active` is false]).

I think that overall the refactor to keep it again in the segment wouldn't change anything in the functionality, but maybe it would be more "correct" 👍 So it would look like:
- For Stacked Column Chart and Stacked Area Chart we have a `calculation` property inside `segment`
- For every chart we have `calculation` property in `interactiveFiltersConfig` with `active`: boolean and `type`: "identity" | "percent"

In fact when I'm looking at the above proposal, it seems that we would duplicate the information inside `segment` and `interactiveFiltersConfig`. Maybe we can sync about this, let me know what do you think 👀

Reply:
The current approach for reference:
- we keep the information about calculation in `interactiveFiltersConfig` (static config that is only changeable while editing the chart). The calculation type is "identity" for every chart, except for stacked charts that could use "percent",
- the dynamic part is done via `InteractiveFiltersState`, which overrides the properties from `interactiveFiltersConfig` for each end user.

Reply:
Also, I just remembered, when the `calculation` was an existing property only for stacked mode in column chart and was removed for grouped chart, it made the radio buttons not checked at all in the screenshot sent here: https://github.com/visualize-admin/visualization-tool/pull/1105#issuecomment-1665496338

Reply:
Decision: keep the calculation type in `segment` (or think again why it should be in `Y`), remove the calculation type from `interactiveFiltersConfig`. Also, duplicate the section to select the type in the `Y` field.